From 22d0c9fb0b3eee8601fc7a23611c7c275f1cc9e0 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Tue, 16 Jul 2013 20:45:12 -0400
Subject: [PATCH] Invalidate Metadata on reload

Metadata is not meant to persist reload as things break badly with non primitive types
This will invalidate metadata on reload so it does not crash everything if a plugin uses it.
---
 src/main/java/org/bukkit/craftbukkit/CraftServer.java | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 4305838..a930a19 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -584,6 +584,13 @@ public final class CraftServer implements Server {
             }
             world.spigotConfig.init(); // Spigot
         }
+        // Spigot start
+        for (Plugin plugin : pluginManager.getPlugins()) {
+            entityMetadata.invalidateAll(plugin);
+            worldMetadata.invalidateAll(plugin);
+            playerMetadata.invalidateAll(plugin);
+        }
+        // Spigot end
 
         pluginManager.clearPlugins();
         commandMap.clearCommands();
-- 
1.8.4.4

